library/src/main/java/com/novoda/downloadmanager/UnlinkedDataRemover.java

```diff
@@ -0,0 +1,32 @@
+package com.novoda.downloadmanager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class UnlinkedDataRemover {
+
+    private final DownloadsPersistence downloadsPersistence;
+    private final LocalFilesDirectory localFilesDirectory;
+
+    UnlinkedDataRemover(DownloadsPersistence downloadsPersistence, LocalFilesDirectory localFilesDirectory) {
+        this.downloadsPersistence = downloadsPersistence;
+        this.localFilesDirectory = localFilesDirectory;
+    }
+
+    void remove() {
+        List<String> filesInV2Database = new ArrayList<>();
+        for (DownloadsBatchPersisted batch : downloadsPersistence.loadBatches()) {
```

Comment:
Surprised you didn't leverage an SQL query to do most of the heavy lifting here, any particular reason? 

"SELECT * FROM RoomFile"

Reply:
@zegnus said in Slack that we shouldn't add any more methods to the `DownloadsPersistence` interface.

Reply:
Really @zegnus ? Can't we add this?